web/src/app/monitors/monitor-create-edit/monitor-create-edit.component.ts

```diff
@@ -93,20 +94,36 @@ export class MonitorCreateEditComponent implements OnInit, OnDestroy {
    * @param monitors monitors list to be updated
    *
    */
-  saveMonitor(uid: string, monitors: MonitorModel[]): void {
-    this.saveMonitorSubscription = this.monitorService.saveMonitors(uid, monitors)
-      .subscribe(
-        () => this.router.navigate([`/projects/${uid}/monitors`]),
-        (error: any): any => this.snackBar.open(error.message, undefined, { duration: 5000 })
-      );
+  saveMonitor(uid: string, monitors: MonitorModel[], monitorUid: string): void {
+    if (monitorUid) {
```

Comment:
I don't think we need the `monitorUid` because it is available in `this.monitorUid`?

Reply:
I think this would be simpler if we moved the `automatic` ping to the service `saveMonitors`, then the if/else is no longer required here?

Reply:
> I don't think we need the `monitorUid` because it is available in `this.monitorUid`?

Fixed

Reply:
> I think this would be simpler if we moved the `automatic` ping to the service `saveMonitors`, then the if/else is no longer required here?

We need to call the `pingMonitor()` with type `automatic` when monitor is updated so I think it should be added here in `saveMonitors()subscribe()`.

Reply:
Fixed and called the `pingMonitor` for `automatic update` inside the service